test/phoenix_html/safe_test.exs

```diff
@@ -61,4 +61,15 @@ defmodule Phoenix.HTML.SafeTest do
 
     assert Safe.to_iodata(datetime) == "2000-01-01T12:13:14+00:30"
   end
+
+  test "impl for URI" do
+    uri = %URI{scheme: "http", host: "www.example.org", path: "/foo", query: "secret=<a&b>"}
+
+    expected = [
+      [[[], "http://www.example.org/foo?secret=" | "&lt;"], "a" | "&amp;"],
+      "b" | "&gt;"
+    ]
+
+    assert Safe.to_iodata(uri) == expected
```

Comment:
would it make sense to pass it through IO.iodata_to_binary to make it more readable?

```elixir
assert uri |> Safe.to_iodata() |> IO.iodata_to_binary() ==
         "http://www.example.org/foo?secret=&lt;a&amp;b&gt;"
```